core/txpool/validation.go

```diff
@@ -170,20 +169,11 @@ func validateBlobSidecar(hashes []common.Hash, sidecar *types.BlobTxSidecar) err
 	if len(sidecar.Blobs) != len(hashes) {
 		return fmt.Errorf("invalid number of %d blobs compared to %d blob hashes", len(sidecar.Blobs), len(hashes))
 	}
-	if len(sidecar.Commitments) != len(hashes) {
-		return fmt.Errorf("invalid number of %d blob commitments compared to %d blob hashes", len(sidecar.Commitments), len(hashes))
-	}
 	if len(sidecar.Proofs) != len(hashes) {
 		return fmt.Errorf("invalid number of %d blob proofs compared to %d blob hashes", len(sidecar.Proofs), len(hashes))
 	}
-	// Blob quantities match up, validate that the provers match with the
-	// transaction hash before getting to the cryptography
-	hasher := sha256.New()
-	for i, vhash := range hashes {
-		computed := kzg4844.CalcBlobHashV1(hasher, &sidecar.Commitments[i])
-		if vhash != computed {
-			return fmt.Errorf("blob %d: computed hash %#x mismatches transaction one %#x", i, computed, vhash)
-		}
```

Comment:
Would be good to add a comment here that says why we aren't validating the commitments here.

Reply:
So we retain validation of the commitments below this.  I just break the logic for validating the hashes into its own method so that it can be used in the handler without requiring a dependency on the txpool package.

Note that with this change, we will be validating the hashes twice for blob transactions received from the network:  once in the handler, and again in the txpool.  This is because submission of local transactions does not go through the handler path.

I think it should be a pretty minor performance hit overall (if even measurable).

Reply:
Ahh, sorry. I misread that.